TZStackView/TZStackView.swift

```diff
@@ -194,22 +202,29 @@ public class TZStackView: UIView {
     }
     
     private func addHiddenListener(view: UIView) {
-        view.addObserver(self, forKeyPath: "hidden", options: [.New], context: &kvoContext)
+        view.addObserver(self, forKeyPath: "hidden", options: [.Old, .New], context: &hiddenKVOContext)
         registeredKvoSubviews.append(view)
     }
     
     private func removeHiddenListener(view: UIView) {
         if let index = registeredKvoSubviews.indexOf(view) {
-            view.removeObserver(self, forKeyPath: "hidden", context: &kvoContext)
+            view.removeObserver(self, forKeyPath: "hidden", context: &hiddenKVOContext)
             registeredKvoSubviews.removeAtIndex(index)
         }
     }
 
     public override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?, change: [String : AnyObject]?, context: UnsafeMutablePointer<Void>) {
-        if let view = object as? UIView where keyPath == "hidden" {
+        if context == &hiddenKVOContext {
+            guard let
+                view = object as? UIView,
+                oldHidden = change?[NSKeyValueChangeOldKey] as? Bool,
+                hidden = change?[NSKeyValueChangeNewKey] as? Bool
+                where oldHidden != hidden
```

Comment:
We should not add this check of equality of oldHidden and new hidden, the `testHidingAnimationSetAgainFirstNotHidden` test will fail because of it. This test starts with a state where subviews are not hidden, then set a subview to hidden with animation, and set the subview back to non-hidden at the middle of the animation. The test fails because when we set a subview to hidden with animation, the subview appears to be non-hidden during the animation, and its hidden property is still false in TZStackView's implementation, so when set the hidden property to false at the middle of the animation, we must do something instead of skipping the whole observation method.